lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -2066,17 +2112,28 @@ static bool checkSingleOverride(ValueDecl *override, ValueDecl *base) {
   }
 
   // FIXME: Possibly should extend to more availability checking.
-  if (auto *attr = base->getAttrs().getUnavailable(ctx)) {
-    diagnoseUnavailableOverride(override, base, attr);
+  auto unavailabilityStatusAndAttr =
+      checkOverrideUnavailability(override, base);
+  auto *unavailableAttr = unavailabilityStatusAndAttr.second;
```

Comment:
Nit: If we are in c++ 17 we could use [structured binding](https://en.cppreference.com/w/cpp/language/structured_binding) here =]
```suggestion
  auto [status, unavailableAttr] =
      checkOverrideUnavailability(override, base);
```

Reply:
That does seem nicer. I'm not going to revise this solely for that right now but I'll keep an eye out for opportunities to improve readability with this syntax in the future now that the project uses C++17.